terraform/table_terraform_module.go

```diff
@@ -0,0 +1,246 @@
+package terraform
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/zclconf/go-cty/cty/gocty"
+	ctyjson "github.com/zclconf/go-cty/cty/json"
+
+	"github.com/Checkmarx/kics/pkg/model"
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+)
+
+func tableTerraformModule(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "terraform_module",
+		Description: "Terraform module information.",
+		List: &plugin.ListConfig{
+			ParentHydrate: tfConfigList,
+			Hydrate:       listModules,
+			KeyColumns:    plugin.OptionalColumns([]string{"path"}),
+		},
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "Module name.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "module_source",
+				Description: "Module source",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "version",
+				Description: "Module version",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "variables",
```

Comment:
```suggestion
				Name:        "arguments",
```

@rollwagen Should this column be named `variables` or `arguments`?

According to https://developer.hashicorp.com/terraform/language/modules/syntax#calling-a-child-module:
> Within the block body (between { and }) are the arguments for the module. Module calls use the following kinds of arguments:
> 
> The source argument is mandatory for all modules.
> 
> The version argument is recommended for modules from a registry.
> 
> Most other arguments correspond to [input variables](https://developer.hashicorp.com/terraform/language/values/variables) defined by the module. (The servers argument in the example above is one of these.)
> 
> Terraform defines a few other meta-arguments that can be used with all modules, including for_each and depends_on.

So in the example above, even though `servers` corresponds to an input variable, it's still an argument in the module.

Thoughts?

Reply:
Good catch & I agree. When you create a module you define _input variables_. However, when you use (or 'call') a module, you pass in _arguments_.